### PR TITLE
Fix PhpArrayDumper notice and PHP 5.3 compatibility

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

**Comment by fabianx-ai** [comment]
Oh wow, when would that not be an object?

**Comment by fabianx-ai** [comment]
It should be possible to unit test this?

Could we setup a scenario for that?

**Comment by drupol** [comment]
Actually, there's nothing to test because:

```
$this->getServiceAliasDefinition($id);
```

Always returns an array, see: https://github.com/LionsAd/service_container/blob/7.x-1.x/lib/Drupal/Core/DependencyInjection/Dumper/PhpArrayDumper.php#L234

File: tests/src/DependencyInjection/Dumper/PhpArrayDumperTest.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  /**
+   * @covers ::dump
+   */
+  public function test_dump() {
+    $dump = unserialize($this->dumper->dump());
+
+    $this->assertTrue(is_array($dump));
+    $this->assertEquals($dump['services']['bar'], 'foo');
```
Comment (fabianx-ai) [inline tests/src/DependencyInjection/Dumper/PhpArrayDumperTest.php]: Hm, but that should not be true.

It should actually be an array with [ 'alias': 'foo' ] set ...

Why do we need to change that?

File: lib/Drupal/Core/DependencyInjection/Dumper/PhpArrayDumper.php [hunk 2]
```diff
@@ -84,7 +81,8 @@ protected function getServiceDefinitions()
       while (isset($aliases[(string) $id])) {
         $id = $aliases[(string) $id];
       }
-      $services[$alias] = (string) $this->getServiceAliasDefinition($id);
+      $alias_definition = $this->getServiceAliasDefinition($id);
+      $services[$alias] = (string) $alias_definition['alias'];
```
Comment (fabianx-ai) [inline lib/Drupal/Core/DependencyInjection/Dumper/PhpArrayDumper.php]: Is that not wrong then?

**Pull request merged**